docs/Design/List.md

```diff
@@ -0,0 +1,106 @@
+# List

```

Comment:
Also: I'd prefer it if you moved this into docs/Design/Lists/{filename} so that later Lists.md can be next to it.